src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1689,7 +1705,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                     ImmutableDictionary<Symbol, Symbol> remappedSymbols = null;
                     var compilation = bodyBinder.Compilation;
                     var isSufficientLangVersion = compilation.LanguageVersion >= MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion();
-                    if (compilation.IsNullableAnalysisEnabled)
+                    if (compilation.IsNullableAnalysisEnabledIn(syntaxNode))
```

Comment:
>syntaxNode [](start = 64, length = 10)

It feels like this is going to include too much for a simple program entry point method and for record's primary constructor.